utt/utt-client-api/src/User.cpp

```diff
@@ -316,11 +316,25 @@ std::unique_ptr<User> User::recoverFromStorage(std::shared_ptr<IStorage> storage
     if (c.getType() == libutt::api::Coin::Normal) {
       user->pImpl_->coins_.emplace(c.getNullifier(), c);
     } else if (c.getType() == libutt::api::Coin::Budget) {
+      // Currently the original budget coin and change budget coins would
+      // be stored in assets.
+      // Until budget coin pool is supported, use the budget with lowest value
+      // to stick to current single outstanding budget workflow model
       if (user->pImpl_->budgetNullifiers_.size() >= 1) {
-        throw std::runtime_error("Currently multiple budget coins are not supported");
+        if (c.getVal() < user->pImpl_->budgetCoin_.value().getVal()) {
```

Comment:
what if the former budget coin value is 0 (which means that we have to mint a new budget coin), will it not be supported at the moment?

Reply:
assumed the size would be zero if there is no budget coin associated is that not the case?

Reply:
okay the scenario is if user spends all his budget then the value is 0